extensions/profiler/client/src/dialogs/profilerCreateSessionDialog.ts

```diff
@@ -0,0 +1,116 @@
+	public readonly onSuccess: vscode.Event<CreateSessionData> = this._onSuccess.event;
+
+
+	constructor(ownerUri: string, templates: Array<sqlops.ProfilerSessionTemplate>) {
+		if (typeof (templates) === 'undefined' || templates === null) {
+			throw new Error(localize('createSessionDialog.templatesInvalid', "Invalid templates list, cannot open dialog"));
+		}
+		if (typeof (ownerUri) === 'undefined' || ownerUri === null) {
+			throw new Error(localize('createSessionDialog.dialogOwnerInvalid', "Invalid dialog owner, cannot open dialog"));
+		}
+		this.model = new CreateSessionData(ownerUri, templates);
+	}
+
+	public async showDialog() {
+		this.dialog = sqlops.window.modelviewdialog.createDialog(this.DialogTitle);
+		this.initializeContent();
+		this.dialog.okButton.onClick(async () => await this.execute());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+		this.dialog.okButton.label = this.CreateButtonText;
+		this.dialog.cancelButton.label = this.CancelButtonText;
+
+		sqlops.window.modelviewdialog.openDialog(this.dialog);
+	}
+
+	private initializeContent() {
+		this.dialog.registerContent(async view => {
+			this.templatesBox = view.modelBuilder.dropDown()
+				.withProperties({
+					values: []
+				}).component();
+
+			this.sessionNameBox = view.modelBuilder.inputBox()
+				.withProperties({
+					required: true,
+					multiline: false,
+					value: ''
+				}).component();
+
+			let formModel = view.modelBuilder.formContainer()
+				.withFormItems([{
+					components: [{
+						component: this.templatesBox,
+						title: 'Select session template:'
+					},
+					{
+						component: this.sessionNameBox,
+
+						title: 'Enter session name:'
+					}],
+					title: this.DialogTitleText
+				}]).withLayout({ width: '100%' }).component();
+
+			await view.initializeModel(formModel);
+
+			if (this.model.templates) {
+				this.templatesBox.values = this.model.getTemplateNames();
+				this.templatesBox.onValueChanged(() => {
+					if (this.sessionNameBox.value === '') {
+						this.sessionNameBox.value = this.templatesBox.value.toString();
+					}
+				});
+			}
+
+			this.sessionNameBox.onTextChanged(() => {
+				if (this.sessionNameBox.value.length > 0) {
+					this.model.sessionName = this.sessionNameBox.value;
+					this.dialog.okButton.enabled = true;
+				} else {
+					this.dialog.okButton.enabled = false;
+				}
+			});
+		});
+	}
+
+	private async execute() {
+		let currentConnection = await sqlops.connection.getCurrentConnection();
+		let profilerService = sqlops.dataprotocol.getProvider<sqlops.ProfilerProvider>(currentConnection.providerName, sqlops.DataProviderType.ProfilerProvider);
+
+		let name = this.sessionNameBox.value;
+		let selected = this.templatesBox.value.toString();
+		let temp = this.model.selectTemplate(selected);
+		profilerService.createSession(this.model.ownerUri, this.sessionNameBox.value, temp);
+	}
+
+	private async cancel() {
```

Comment:
Does there need to be an implementation here? Otherwise there's no need for the method at all